File: 215. Kth Largest Element in an Array/findKthLargest.py
from heapq import *
class Solution(object):
    # def findKthLargest(self, nums, k):
    #     """
    #     :type nums: List[int]
    #     :type k: int
    #     :rtype: int
    #     """
    #     nums.sort(key = lambda x: -x)
    #     return nums[k-1]
        
    def findKthLargest(self, nums, k):
        """
        :type nums: List[int]
        :type k: int
        :rtype: int
        """
        output = 0
        heap = []
        for n in nums:
            heappush(heap, -n)
        for i in range(k):
            output = heappop(heap)
        return -output
        

File: 215. Kth Largest Element in an Array/findKthLargest.java
class Solution {
    //use a min-heap with size k.
    //After checking all the objs in the given array, we output the first element in the heap.
    public int findKthLargest(int[] nums, int k) {
        PriorityQueue<Integer> heap = new PriorityQueue<>();
        for(int i : nums){
            heap.offer(i);
            if(heap.size() > k){
                heap.poll();
            }
        }
        return heap.peek();
    }
}